src/jobs/attestationRequiredNotification.ts

```diff
@@ -0,0 +1,105 @@
+import { chunk } from 'lodash';
+
+import logger from '../logger';
+import { AppConfig, getAppConfig } from '../config';
+import { NOTIFICATION_UNIT_OF_TIME, REQUEST_CHUNK_SIZE } from '../utils/constants';
+import { ApplicationDocument, ApplicationModel } from '../domain/model';
+import { Application } from '../domain/interface';
+import { buildReportItem, getEmptyReport } from './utils';
+import { sendAttestationRequiredEmail } from '../domain/service';
+import { getDayRange } from '../utils/calculations';
+
+export const JOB_NAME = 'ATTESTATION REQUIRED NOTIFICATIONS';
+
+// Check + notification for applications entering attestation period
+async function attestationRequiredNotificationCheck(
+  currentDate: Date,
+  emailClient: Transporter<SMTPTransport.SentMessageInfo>,
+) {
+  const config = await getAppConfig();
+  try {
+    logger.info(`${JOB_NAME} - Initiating...`);
+    const report = await getAttestableNotificationReport(currentDate, emailClient, config);
+    logger.info(`${JOB_NAME} - Completed.`);
+    if (report?.errors.length) {
+      logger.warn(`${JOB_NAME} - Completed, with errors.`);
+    }
+    logger.info(`${JOB_NAME} - Returning report.`);
+    return report;
+  } catch (err) {
+    logger.error(`${JOB_NAME} - Failed to complete, with error: ${(err as Error).message}`);
+    return `${JOB_NAME} - Failed to complete, with error: ${(err as Error).message}`;
+  }
+}
+
+const getAttestableNotificationReport = async (
+  currentDate: Date,
+  emailClient: Transporter<SMTPTransport.SentMessageInfo>,
+  config: AppConfig,
+) => {
+  const query = getAttestableQuery(config, currentDate);
+  const attestableAppCount = await ApplicationModel.find(query).countDocuments();
+  if (attestableAppCount === 0) {
+    logger.info(`${JOB_NAME} - No applications are entering the attestation period.`);
+    logger.info(`${JOB_NAME} - Generating report.`);
+    return getEmptyReport();
+  }
+
+  logger.info(
+    `${JOB_NAME} - ${attestableAppCount} applications are entering the attestation period.`,
+  );
+  const attestableApps = await ApplicationModel.find(query).exec();
+  const apps: Application[] = attestableApps.map((app: ApplicationDocument) => {
+    logger.info(`${JOB_NAME} - Should notify ${app.appId} regarding attestation.`);
+    return app.toObject() as Application;
+  });
+
+  // is this waiting for each email chunk like i'm expecting?
+  const emails = chunk(apps, REQUEST_CHUNK_SIZE).map(async (appChunk: Application[]) => {
+    logger.info(`${JOB_NAME} - Initiating email requests.`);
+    return Promise.allSettled(
+      appChunk.map(async (app) => sendAttestationRequiredEmail(app, config, emailClient)),
+    );
+  });
+
+  const results: PromiseSettledResult<any>[][] = [];
+  for (const email of emails) {
+    const result = (await email) as PromiseSettledResult<any>[];
+    results.push(result);
+  }
+  const allResults = results.flat();
+  logger.info(`${JOB_NAME} - Generating report.`);
```

Comment:
the code block I suggested for logging includes this change.

Reply:
the other option is, running any of these jobs separately, each job returns its own report so that could be logged/returned as needed, right?

Reply:
Each job returning its own report is good. I would probably have each job log itself instead of relying on whatever triggers them to log the report, but so long as there is consistency in the system thats fine by me.

Reply:
I added logs for successful and failed job reports, as well as returning the report obj